File: easy/leetcode14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        ans = ""
        strs_lens = len(strs)
        if strs_lens == 0:
            return ""
        if strs_lens == 1:
            return strs[0]
        else:
            for i in range(min(len(strs[0]),len(strs[1]))):
                if strs[0][i] == strs[1][i]:
                    ans += strs[0][i]
                else:
                    break
            #到这里就求出了前两位的公共字串
            for i in range(2, strs_lens):
                if len(strs[i]) == 0:
                    return ""
                lensx = min(len(ans),len(strs[i]))
                for j in range(lensx):
                    if strs[i][j] == ans[j]:
                        if lensx < len(ans):
                            ans = ans[:lensx]
                        continue
                    else:
                        ans = ans[:j]
                        break
        if ans == "":
            return ""
        else:
            return ans

File: easy/leetcode14.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0 || (strs.length == 1 && Objects.equals(strs[0], ""))) {
            return "";
        }
        StringBuilder res = new StringBuilder();
        for (int i = 0; i < 100; i++) {
            if (strs[0].length() <= i) {
                break;
            }
            res.append(strs[0].charAt(i));
            for (int j = 0; j < strs.length; j++) {
                String tempString = strs[j];
                if (tempString.length() < i+1) {
                    return res.delete(i, i+1).toString();
                }
                if (!tempString.substring(0, i+1).equals(res.toString())) {
                    return res.delete(i, i+1).toString();
                }
            }
        }
        return res.toString();
    }
}